src/units.cpp

```diff
@@ -157,40 +167,65 @@ void Units::setBaseUnit(bool state)
     mPimpl->mBaseUnit = state;
 }
 
-void Units::addUnit(const std::string & units, PREFIXES prefix, double exponent,
+void Units::addUnit(const std::string &name, const std::string &prefix, double exponent,
              double multiplier, double offset)
 {
     Unit u;
-    u.mUnits = units;
-    u.mPrefixEnum = prefix;
-    u.mExponent = exponent;
-    u.mMultiplier = multiplier;
-    u.mOffset = offset;
-
+    u.mName = name;
+    // Allow all nonzero user-specified prefixes
+    try
+    {
+        double prefixDouble = std::stod(prefix);
+        if (prefixDouble != 0.0) {
+            u.mPrefix = prefix;
+        }
+    } catch(...) {
+        u.mPrefix = prefix;
+    }
+    if (exponent != 1.0) {
+        std::ostringstream strs;
+        strs << exponent;
+        u.mExponent = strs.str();
+    }
+    if (multiplier != 1.0) {
+        std::ostringstream strs;
+        strs << multiplier;
+        u.mMultiplier = strs.str();
+    }
+    if (offset != 0.0) {
+        std::ostringstream strs;
+        strs << offset;
+        u.mOffset = strs.str();
+    }
     mPimpl->mUnits.push_back(u);
 }
 
-void Units::addUnit(const std::string & units, int prefix, double exponent,
+void Units::addUnit(const std::string &name, PREFIXES prefix, double exponent,
              double multiplier, double offset)
 {
-    Unit u;
-    u.mUnits = units;
-    u.mPrefixInt = prefix;
-    u.mExponent = exponent;
-    u.mMultiplier = multiplier;
-    u.mOffset = offset;
+    auto search = prefixToString.find(prefix);
```

Comment:
I did a quick Google re: `find()` vs `count()` and it looks like for this case, `find()` is [slightly preferred](http://stackoverflow.com/questions/25490357/checking-for-existence-in-stdmap-count-vs-find). 

The justification is that `count()` is slightly less efficient as it continues iterating until the end of the map range whether or not the key has been found. The intent of `find()` is also arguably clearer (we are trying to find the key in the map).